packages/core/src/test/shared/utilities/pathFind.test.ts

```diff
@@ -42,4 +41,50 @@ describe('pathFind', function () {
         const regex = /bin[\\\/](code|code-insiders)$/
         assert.ok(regex.test(vscPath), `expected regex ${regex} to match: "${vscPath}"`)
     })
+
+    describe('findSshPath', function () {
+        it('first tries ssh in $PATH', async function () {
+            const workspace = await testutil.createTestWorkspaceFolder()
+            const fakeSshPath = path.join(workspace.uri.fsPath, `ssh${isWin() ? '.cmd' : ''}`)
+
+            await testutil.withEnvPath(workspace.uri.fsPath, async () => {
```

Comment:
if we just set $PATH here directly, can `afterEach` restore it? 

the "with" pattern can be useful but it might be confusing if we have different ways of saving/storing state in tests (but maybe there is a good reason for it)

Reply:
Yeah, this definitely reduces the amount of code as well. My only concern is that adding more behavior into `beforeEach` and `afterEach` can make it hard to isolate the behavior of each test, and sometimes makes it harder to add tests if it doesn't fit with the pattern. 

I think this is still the better route since it is inline with the rest of the test code (ex. sinon, clock).